File: nitpickers/pyql/pyql/ast/form/block.py
from pyql.ast.ast import ASTNode


class Block(ASTNode):

    def __init__(self, location, statements: []):
        super().__init__(location)
        self._statements = statements

    @property
    def statements(self):
        return self._statements

    def __repr__(self):
        return str({"AST Block " + str(self.location): str(self.statements)})

File: DogeQL/src/main/kotlin/doge/ast/node/Block.kt
package doge.ast.node

import doge.ast.location.SourceLocation
import doge.visitor.QuestionnaireASTBaseVisitor

class Block(val statements: List<Statement>, location: SourceLocation) : QLNode(location) {

    override fun <T> accept(visitor: QuestionnaireASTBaseVisitor<T>): T {
        return visitor.visit(this)
    }

}